### DISCUSSION TITLE
Static page generation as a separate command

### DISCUSSION THREAD

**Discussion by gromchen** [vercel/next.js#22947] [Ideas]
### Describe the feature you'd like to request

Currently, we build and deploy around 13 websites with different content and style however with the same code base.
Before static generation, in our pipeline, we nicely build the .next folder once and deployed it with a different .env variable.
With the introduction of static generation, we have to duplicate the build step for every environment, however, I suppose the build step is still the same except for the static page generation step.

### Describe the solution you'd like

It would greatly improve our pipeline speed if we had the static page generation step as a separate command.

### Describe alternatives you've considered

Now, we have to build 13 sites for every commit which consume an extremely excessive amount of resources.

**Comment by vadistic** [comment]
Hey, I'm also very interested in this for a bit other reasons.

I have a statically exported webpage with literally one dynamic route and frequent content updates.

I would love to build actual feature code (styles, component and such) only once, pack the result `.next` dist into quite lightweight cli and then just rebuild `getStaticProps` / `getStaticPaths` with new new data.

I'm quite sure next is quite capable of this since you can build new pages in ISG - the only thing required would this separate command to run it over compiled `.next` dist.
